metomi/rosie/ws_client_auth.py

```diff
@@ -33,23 +34,13 @@
 from metomi.rose.resource import ResourceLocator
 
 try:
-    from gi import require_version, pygtkcompat
-
-    require_version('Gtk', '3.0')  # For GTK+ >=v3 use PyGObject; v2 use PyGTK
-    require_version('Secret', '1')
-    from gi.repository import Secret
-
-    del pygtkcompat
-    GI_FLAG = True
-except (ImportError, ValueError):
-    GI_FLAG = False
-try:
-    with warnings.catch_warnings():
-        warnings.simplefilter('ignore')
-        import gtk
-    import gnomekeyring
+    import keyring
+    KEYRING_FLAG = True
 except ImportError:
-    pass
+    KEYRING_FLAG = False
+else:
+    import keyring.errors
+    from keyring.backends.fail import Keyring as NoAvailableKeyring
```

Comment:
This is linguistically confusing - and looking at the class docstring makes me wonder why you'd want to import this object. I think a comment would be in order.

```suggestion
    # Fallback non-functional keyring: Used to check that the module.get_keyring()
    # has given us a keyring we can use.
    from keyring.backends.fail import Keyring as NoAvailableKeyring
```